src/meta_memcache/configuration.py

```diff
@@ -1,18 +1,32 @@
 import hashlib
 import socket
-from typing import Callable, NamedTuple
+from typing import Callable, NamedTuple, Optional
 
 from meta_memcache.base.connection_pool import ConnectionPool
 from meta_memcache.protocol import Key
 from meta_memcache.settings import DEFAULT_MARK_DOWN_PERIOD_S
 
 
-class IPPort(NamedTuple):
-    ip: str
+class ServerAddress(NamedTuple):
+    host: str
     port: int
+    # ShardedCachePool with HashRing uses str(Server) to
+    # configure the ring of servers and the shardmap. You can
+    # override this id to control the HashRing configuration,
+    # for example, to mimic bmemcached sharding, you should
+    # user server_id = <host>:<port>_<user>_<password>
+    # If you want to replace servers in-place you can
+    # use numerical server_ids, and change the destination
+    # IP maintaining the server_id for clean swaps.
+    server_id: Optional[str] = None
 
     def __str__(self) -> str:
-        return f"{self.ip}:{self.port}"
+        if self.server_id is not None:
+            return self.server_id
+        elif ":" in self.host:
+            return f"[{self.host}]:{self.port}"
```

Comment:
Why this path?

Reply:
IPv6 addresses contain semicolons, and the right way of representing the address is `[<ipv6>]:port` to make it non-ambiguous. The ipv6 allows to shorten address, replacing any contiguous tuples of 0 bytes with a single `::`, `2001:0db8:0000:0000:0000:8a2e:0370:7334` becomes `2001:db8::8a2e:370:7334`. If you add a port, it becomes ambiguous `2001:db8::8a2e:370:7334:80` might mean `2001:db8:0000:0000:0000:8a2e:0370:7334` port `80` or `2001:db8:0000:0000:8a2e:0370:7334:0080`, hence, ipv6+port must be represented as `[2001:db8::8a2e:370:7334]:80`

Reply:
Ah ok, didn't know we're supporting IPv6 :D

Reply:
Not explicitly, someone will need to provide a socket open function that opens ipv6 (which doesn't by default), but nothing will be any different, so I want it to work if someone defines their own socket connect for ipv6.